pkg/segment/aggregations/generateevents.go

```diff
@@ -86,3 +99,112 @@ func PerformGenTimes(aggs *structs.QueryAggregators) error {
 
 	return nil
 }
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+// Ensure record and column lengths are valid before calling this function.
+func createRecord(columnNames []string, record []string) map[string]interface{} {
+	recordMap := make(map[string]interface{})
+	for i, col := range columnNames {
+		recordMap[col] = record[i]
+	}
+	return recordMap
+}
+
+func PerformInputLookup(aggs *structs.QueryAggregators) error {
+	var reader *csv.Reader
+	if aggs.GenerateEvent.InputLookup == nil {
+		return fmt.Errorf("PerformInputLookup: InputLookup is nil")
+	}
+	filename := aggs.GenerateEvent.InputLookup.Filename
+
+	if !checkCSVFormat(filename) {
+		return fmt.Errorf("PerformInputLookup: Only CSV format is currently supported")
```

Comment:
I think this will be a bit more informative:
```suggestion
		return fmt.Errorf("PerformInputLookup: Only .csv and .csv.gz formats are currently supported")
```

Reply:
addressed in the latest commit.